File: TheaterSuggestions/Java/ScreeningSeatingArrangementSuggestions/src/main/java/org/weaveit/seatingplacesuggestions/SuggestionIsMade.java
package org.weaveit.seatingplacesuggestions;

import java.util.List;

public record SuggestionIsMade(
        List<SeatingPlace> suggestedSeats,
        int partyRequested,
        PricingCategory pricingCategory
) {
    public SuggestionIsMade {
        suggestedSeats = List.copyOf(suggestedSeats);
    }

    public SuggestionIsMade(SeatingOption seatingOption) {
        this(seatingOption.seats(), seatingOption.partyRequested(), seatingOption.pricingCategory());
    }

    public List<String> seatNames() {
        return suggestedSeats.stream().map(SeatingPlace::toString).toList();
    }

    public boolean matchExpectation() {
        return suggestedSeats.size() == partyRequested;
    }
}

File: TheaterSuggestions/Kotlin/ScreeningSeatingArrangementSuggestions/src/main/kotlin/org/weaveit/seatingplacesuggestions/SuggestionIsMade.kt
package org.weaveit.seatingplacesuggestions

class SuggestionIsMade(seatingOptionIsSuggested: SeatingOptionIsSuggested) {
    private val suggestedSeats: List<SeatingPlace> = seatingOptionIsSuggested.seats()
    private val partyRequested: Int = seatingOptionIsSuggested.partyRequested
    val pricingCategory: PricingCategory = seatingOptionIsSuggested.pricingCategory

    fun seatNames(): List<String> {
        return suggestedSeats.map { it.toString() }
    }

    fun matchExpectation(): Boolean {
        return suggestedSeats.size == partyRequested
    }
}
